File: kotlin-pr/50/kotlin/code-quality-logging/PaymentProcessor.kt
class PaymentProcessor {
    fun processPayment(cardNumber: String, cvv: String, amount: Double): Boolean {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        println("Processing payment: Card=$cardNumber, CVV=$cvv, Amount=$amount")
        
        return try {
            // Payment processing logic
            println("Payment successful: $cardNumber charged $$amount")
            true
        } catch (ex: Exception) {
            println("Payment failed for card $cardNumber: ${ex.message}")
            false
        }
        // {/fact}
    }
}

File: php-pr/51/php/code-quality-logging/PaymentProcessor.php
<?php

class PaymentProcessor
{
    public function processPayment($cardNumber, $cvv, $amount)
    {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        error_log("Processing payment: Card={$cardNumber}, CVV={$cvv}, Amount={$amount}");
        
        try {
            // Payment processing logic
            error_log("Payment successful: {$cardNumber} charged \${$amount}");
            return true;
        } catch (Exception $e) {
            error_log("Payment failed for card {$cardNumber}: " . $e->getMessage());
            return false;
        }
        // {/fact}
    }
}